### PR TITLE
Add stream logs to prod

### PR DESCRIPTION
**Pull request by IvanPombo**
This PR makes the minor changes to adequately introduce the streaming of logs into Production mode.

Results:
Screenshot 2024-01-26 at 11 04 07

## QA

Setup your environment to use the dev api. Export the env variable INDUCTIVA_TASK_LOGS_URL=wss://logs-dev.inductiva.ai.
Launch a simulation, retrieve the task ID and in another terminal check inductiva logs <task_id> is working correctly.

### CODE REVIEW

**Review by Pbarbosa92 — APPROVED** [review]

**Comment by Pbarbosa92** [comment]
I get this error
Captura de ecrã 2024-01-26, às 11 21 20

**Comment by IvanPombo** [comment]
> I get this error Captura de ecrã 2024-01-26, às 11 21 20

Can you tell me your environment variables? You need to use your api-key from dev.

**Comment by Pbarbosa92** [comment]
> > I get this error Captura de ecrã 2024-01-26, às 11 21 20
> 
> Can you tell me your environment variables? You need to use your api-key from dev.

I just changed to an api key from dev and I have the same problem.

File: inductiva/tasks/streams.py [hunk 1]
```diff
@@ -51,20 +54,26 @@ def __init__(self,
...
         return websocket.WebSocketApp(self.websocket_url,
+                                      header={"X-API-Key": inductiva.api_key},
                                       on_open=self.__on_open,
                                       on_error=self.__on_error,
                                       on_close=self.__on_close,
                                       on_message=self.__on_message)
 
     def __on_message(self, unused_ws, message):
-        print(message, file=self.fout)
+        data = json.loads(message)
```
Comment (sergiomsantos) [inline inductiva/tasks/streams.py:64]: it would be useful to have a schema of the message's json.

**Review by sergiomsantos — APPROVED** [review]

File: inductiva/tasks/streams.py (hunk 1 above)
Comment (IvanPombo) [reply inductiva/tasks/streams.py:64]: Indeed! Maybe in the next steps we can provide that.

**Comment by AugustoPeres** [comment]
### QA assured ✅ 

Launched a task using this script:

```
import inductiva

# Create a MachineGroup object with a single machine of the desired type
mg = inductiva.resources.MachineGroup(
    machine_type="c2-standard-16",
    num_machines=1,
    disk_size_gb=70,
)
mg.start()

input_dir = inductiva.utils.files.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "splishsplash-input-example.zip",
    unzip=True)

# Set simulation input directory
splishsplash = inductiva.simulators.SplishSplash()

task = splishsplash.run(input_dir=input_dir,
                        on=mg,
                        sim_config_filename="config.json")

task.wait()
task.download_outputs()
```

Set all the required env variables.
From the logs I got the output:

```
Registering machine group configurations:
> Name: api-m3eydmsukpxymo9pqp4ri0jad
> Machine Type: c2-standard-16
> Disk size: 70 GB
> Number of machines: 1
> Spot: False
Estimated cloud cost for all machines : 0.919 $/h
Starting machine group. This may take a few minutes.
Note that stopping this local process will not interrupt the creation of the machine group. Please wait...
Machine group successfully started in 0:00:20.
Downloading from url https://storage.googleapis.com/inductiva-api-demo-files/splishsplash-input-example.zip to the local path: /Users/augustoperes/Documents/inductiva_test_scripts/splishsplash-input-example.zip
Task Information: 
> ID: 332zvobp97fh8yec5krq3zmpr
> Method: sph.splishsplash.run_simulation
> Local input directory: /Users/augustoperes/Documents/inductiva_test_scripts/splishsplash-input-example
> Computational resources:
  >> Submmiting to the Machine Group: api-m3eydmsukpxymo9pqp4ri0jad
  >> Resource type: c2-standard-16None
Preparing local input directory /Users/augustoperes/Documents/inductiva_test_scripts/splishsplash-input-example (160 B) for upload
Local input directory sucessfully uploaded.
Task 332zvobp97fh8yec5krq3zmpr submitted to the queue of the Machine Group: api-m3eydmsukpxymo9pqp4ri0jad.
Task configuration metadata is saved in a file in the local input directory /Users/augustoperes/Documents/inductiva_test_scripts/splishsplash-input-example/uploaded_metadata.json and added to the general tasks metadata file in /Users/augustoperes/Documents/inductiva_test_scripts/task_metadata.json.
Consider tracking the status of the task via cli: $inductiva tasks list
Task 332zvobp97fh8yec5krq3zmpr successfully queued and waiting to picked-up for execution...
```

Using `inductiva logs 332zvobp97fh8yec5krq3zmpr` I get:
```
Websocket connected
Opening socket connection to logs of task 332zvobp97fh8yec5krq3zmpr ...
# COMMAND: ['/SPlisHSPlasH_CPU/bin/SPHSimulator', '/opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/sim_dir/config.json', '--output-dir', '/opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/output/artifacts', '--no-gui']
SPlisHSPlasH version: 2.13.1
Output directory: /opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/output/artifacts
Load scene file: /opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/sim_dir/config.json
Initialize static boundary simulation
Initialize fluid particles
Volume sampling of /opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/sim_dir/unit_box.obj
Loading /opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/sim_dir/unit_box.obj
Number of triangles: 12
Number of vertices: 8
SDF resolution: 20, 20, 20
time SDF Generation: 11.2613 ms
Save particle sampling: /opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/sim_dir/Cache/unit_box_fluid_0.05000000074505805969_i0_m0_s0.5_0.5_0.5_r20_20_20.bgeo
Number of fluid particles: 125
Loading /opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/sim_dir/unit_box.obj
Number of triangles: 12
Number of vertices: 8
Poisson disk surface sampling of /opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/sim_dir/unit_box.obj
Save particle sampling: /opt/executer-tracker/workdir/332zvobp97fh8yec5krq3zmpr/sim_dir/Cache/unit_box_sb_0.05000000074505805969_1_1_1_m0.bgeo
Number of boundary particles: 1457
Initialize boundary volume
Sum of emitted particles: 0
Sum of reused particles: 0
Average time: SimStep: 0.578826 ms
Average time: animateParticles: 0.000148108 ms
Average time: emitParticles: 8.63919e-05 ms
Average time: pressureSolve: 0.0330445 ms
Average time: computeNonPressureForces: 0.0198004 ms
Average time: CG solve: 0.00620095 ms
Average time: divergenceSolve: 0.0268063 ms
Average time: computeDFSPHFactor: 0.00497558 ms
Average time: precomputeValues: 0.00641304 ms
Average time: neighborhood_search: 0.471593 ms
Average time: Poisson disk sampling: 12.0287 ms
Average time: Volume sampling: 16.6205 ms
---------------------------------------------------------------------------

Time sum: SimStep: 5699.7 ms
Time sum: animateParticles: 1.45842 ms
Time sum: emitParticles: 0.850701 ms
Time sum: pressureSolve: 325.389 ms
Time sum: computeNonPressureForces: 194.975 ms
Time sum: CG solve: 61.0608 ms
Time sum: divergenceSolve: 263.961 ms
Time sum: computeDFSPHFactor: 48.9945 ms
Time sum: precomputeValues: 63.1492 ms
Time sum: neighborhood_search: 4643.78 ms
Time sum: Poisson disk sampling: 12.0287 ms
Time sum: Volume sampling: 16.6205 ms
---------------------------------------------------------------------------

Average number: Time step size: 0.000100554
Average number: DFSPH - iterations: 2.00691
Average number: Visco iterations: 0
Average number: DFSPH - iterationsV: 1.01127
---------------------------------------------------------------------------

Total number: Time step size: 0.990156
Total number: DFSPH - iterations: 19762
Total number: Visco iterations: 0
Total number: DFSPH - iterationsV: 9958
---------------------------------------------------------------------------

Closed stream with 1011: all ingesters closed the connection.
```

**Pull request merged**